orangecontrib/text/bagofowords.py

```diff
@@ -1,53 +1,126 @@
-        self.progress_callback = progress_callback
-        self.error_callback = error_callback
-        self.dictionary = None
-
-    def __call__(self, corpus, use_tfidf=False):
-        if corpus is None:
-            raise ValueError(
-                    'Cannot compute Bag of Words without an input corpus.'
-            )
-
-        self.check_progress()  # Step 1
-
-        self.dictionary = corpus.dictionary
-
-        # Term frequencies.
-        bag_of_words = [self.dictionary.doc2bow(i) for i in corpus.tokens]
-
-        self.check_progress()  # Step 2
-
-        if use_tfidf:
-            tfidf_model = TfidfModel(bag_of_words)
-            bag_of_words = tfidf_model[bag_of_words]
+from sklearn.feature_extraction import text
+
+from orangecontrib.text.corpus import Corpus
 
-        self.check_progress()  # Step 3
 
-        X = matutils.corpus2csc(
-                bag_of_words,
-                num_terms=len(self.dictionary.keys()),
-                dtype=np.float64
-        ).T
+__all__ = ['CountVectorizer', 'TfidfVectorizer', 'SimhashVectorizer', 'Simhash']
 
-        bow_corpus = corpus.copy()
-        feats = [v for k, v in sorted(self.dictionary.items())]
-        bow_corpus.extend_attributes(X, feats, var_attrs={'hidden': True})
 
-        self.check_progress()  # Step 4
+class SklearnVectorizerMixin:
+
+    def fit(self, data, y=None):
+        if isinstance(data, Corpus):
+            return super().fit(data.tokens, y)
+        return super().fit(data, y)
+
+    def fit_transform(self, corpus, y=None):
+        if isinstance(corpus, Corpus):
+            matrix = super().fit_transform(corpus.tokens, y)
+            return self.__extend_corpus(corpus, matrix)
+        return super().fit_transform(corpus, y)
 
-        return bow_corpus
+    def transform(self, corpus):
+        if isinstance(corpus, Corpus):
+            matrix = super().transform(corpus.tokens)
+            return self.__extend_corpus(corpus, matrix)
+        return super().transform(corpus)
+
+    def __extend_corpus(self, corpus, matrix):
+        new_corpus = corpus.copy()
+        new_corpus.extend_attributes(matrix, sorted(self.vocabulary_.keys()),
+                                     var_attrs={'hidden': True})
+        return new_corpus
+
+
+class CountVectorizer(SklearnVectorizerMixin, text.CountVectorizer):
+    name = 'Count Vectorizer'
+
+    def __init__(self, binary=False, ngram_range=(1, 1)):
+        super().__init__(input='content', encoding='utf-8',
+                         decode_error='strict', strip_accents=None,
+                         lowercase=False, preprocessor=lambda x: x, tokenizer=lambda x: x,
+                         stop_words=None, token_pattern=r"(?u)\b\w\w+\b",
+                         ngram_range=ngram_range, analyzer='word',
+                         max_df=1.0, min_df=1, max_features=None,
+                         vocabulary=None, binary=binary, dtype=np.int64)
```

Comment:
I think gensim would be much more appropriate here. I will write more about this in a bit. But generally it seems like you had to take special care to really disable all the features scikit provides since we do it manually.  Do you really prefer this instead of something like:

```
corpus = [dictionary.doc2bow(text) for text in corpus.tokens]
gensim.matutils.corpus2csc(corpus, dtype=int).T
```

Tfidf could be done similarly: 
```
corpus = [dictionary.doc2bow(text) for text in corpus.tokens]
models.TfidfModel(corpus)
gensim.matutils.corpus2csc(corpus, dtype=np.float).T
```